### PR TITLE
[Examples] add Contentful example

### PR DESCRIPTION
**Pull request by lfades**

### CODE REVIEW

**Comment by ijjk** [comment]
# Stats from current PR

<details>
<summary><strong>Default Server Mode</strong> (Increase detected ⚠️)</summary>

<br/>

<details>
<summary><strong>General</strong></summary>

|  | zeit/next.js canary  | lfades/next.js examples/cms-contentful | Change |
| - | - | - | - |
| buildDuration | 12s | 12.1s | ⚠️ +40ms |
| nodeModulesSize | 55.2 MB | 55.2 MB | ✓ |

</details>

<details>
<summary><strong>Page Load Tests</strong> Overall increase ✓</summary>

|  | zeit/next.js canary  | lfades/next.js examples/cms-contentful | Change |
| - | - | - | - |
| / failed reqs | 0 | 0 | ✓ |
| / total time (seconds) | 1.913 | 1.945 | ⚠️ +0.03 |
| / avg req/sec | 1306.84 | 1285.57 | -21.27 |
| /error-in-render failed reqs | 0 | 0 | ✓ |
| /error-in-render total time (seconds) | 1.012 | 0.979 | -0.03 |
| /error-in-render avg req/sec | 2471.46 | 2552.53 | ⚠️ +81.07 |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | zeit/next.js canary  | lfades/next.js examples/cms-contentful | Change |
| - | - | - | - |
| main-HASH.js gzip | 6.27 kB | 6.27 kB | ✓ |
| webpack-HASH.js gzip | 746 B | 746 B | ✓ |
| de003c3a9d30..e8e1.js gzip | 10.3 kB | 10.3 kB | ✓ |
| framework.a1..NSE.txt gzip | 220 B | 220 B | ✓ |
| framework.a1..NSE.txt gzip | 220 B | 220 B | ✓ |
| framework.HASH.js gzip | 39.2 kB | 39.2 kB | ✓ |
| Overall change | 56.9 kB | 56.9 kB | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons) Modern</strong></summary>

|  | zeit/next.js canary  | lfades/next.js examples/cms-contentful | Change |
| - | - | - | - |
| main-HASH.module.js gzip | 4.82 kB | 4.82 kB | ✓ |
| webpack-HASH..dule.js gzip | 746 B | 746 B | ✓ |
| de003c3a9d30..dule.js gzip | 6.88 kB | 6.88 kB | ✓ |
| framework.HA..dule.js gzip | 39.2 kB | 39.2 kB | ✓ |
| Overall change | 51.7 kB | 51.7 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | zeit/next.js canary  | lfades/next.js examples/cms-contentful | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 26.3 kB | 26.3 kB | ✓ |
| Overall change | 26.3 kB | 26.3 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | zeit/next.js canary  | lfades/next.js examples/cms-contentful | Change |
| - | - | - | - |
| _app.js gzip | 1.23 kB | 1.23 kB | ✓ |
| _error.js gzip | 3.15 kB | 3.15 kB | ✓ |
| hooks.js gzip | 663 B | 663 B | ✓ |
| index.js gzip | 222 B | 222 B | ✓ |
| link.js gzip | 2.06 kB | 2.06 kB | ✓ |
| routerDirect.js gzip | 280 B | 280 B | ✓ |
| withRouter.js gzip | 278 B | 278 B | ✓ |
| Overall change | 7.88 kB | 7.88 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages Modern</strong></summary>

|  | zeit/next.js canary  | lfades/next.js examples/cms-contentful | Change |
| - | - | - | - |
| _app.module.js gzip | 598 B | 598 B | ✓ |
| _error.module.js gzip | 2.09 kB | 2.09 kB | ✓ |
| hooks.module.js gzip | 383 B | 383 B | ✓ |
| index.module.js gzip | 223 B | 223 B | ✓ |
| link.module.js gzip | 1.52 kB | 1.52 kB | ✓ |
| routerDirect..dule.js gzip | 279 B | 279 B | ✓ |
| withRouter.m..dule.js gzip | 278 B | 278 B | ✓ |
| Overall change | 5.37 kB | 5.37 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | zeit/next.js canary  | lfades/next.js examples/cms-contentful | Change |
| - | - | - | - |
| _buildManifest.js gzip | 61 B | 61 B | ✓ |
| _buildManife..dule.js gzip | 61 B | 61 B | ✓ |
| Overall change | 122 B | 122 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | zeit/next.js canary  | lfades/next.js examples/cms-contentful | Change |
| - | - | - | - |
| index.html gzip | 925 B | 925 B | ✓ |
| link.html gzip | 936 B | 936 B | ✓ |
| withRouter.html gzip | 924 B | 924 B | ✓ |
| Overall change | 2.79 kB | 2.79 kB | ✓ |

</details>

</details>
<hr/>
<details>
<summary><strong>Serverless Mode</strong></summary>

<br/>

<details>
<summary><strong>General</strong></summary>

|  | zeit/next.js canary  | lfades/next.js examples/cms-contentful | Change |
| - | - | - | - |
| buildDuration | 12.6s | 12.3s | -230ms |
| nodeModulesSize | 55.2 MB | 55.2 MB | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | zeit/next.js canary  | lfades/next.js examples/cms-contentful | Change |
| - | - | - | - |
| main-HASH.js gzip | 6.27 kB | 6.27 kB | ✓ |
| webpack-HASH.js gzip | 746 B | 746 B | ✓ |
| de003c3a9d30..e8e1.js gzip | 10.3 kB | 10.3 kB | ✓ |
| framework.a1..NSE.txt gzip | 220 B | 220 B | ✓ |
| framework.a1..NSE.txt gzip | 220 B | 220 B | ✓ |
| framework.HASH.js gzip | 39.2 kB | 39.2 kB | ✓ |
| Overall change | 56.9 kB | 56.9 kB | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons) Modern</strong></summary>

|  | zeit/next.js canary  | lfades/next.js examples/cms-contentful | Change |
| - | - | - | - |
| main-HASH.module.js gzip | 4.82 kB | 4.82 kB | ✓ |
| webpack-HASH..dule.js gzip | 746 B | 746 B | ✓ |
| de003c3a9d30..dule.js gzip | 6.88 kB | 6.88 kB | ✓ |
| framework.HA..dule.js gzip | 39.2 kB | 39.2 kB | ✓ |
| Overall change | 51.7 kB | 51.7 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | zeit/next.js canary  | lfades/next.js examples/cms-contentful | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 26.3 kB | 26.3 kB | ✓ |
| Overall change | 26.3 kB | 26.3 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | zeit/next.js canary  | lfades/next.js examples/cms-contentful | Change |
| - | - | - | - |
| _app.js gzip | 1.23 kB | 1.23 kB | ✓ |
| _error.js gzip | 3.15 kB | 3.15 kB | ✓ |
| hooks.js gzip | 663 B | 663 B | ✓ |
| index.js gzip | 222 B | 222 B | ✓ |
| link.js gzip | 2.06 kB | 2.06 kB | ✓ |
| routerDirect.js gzip | 280 B | 280 B | ✓ |
| withRouter.js gzip | 278 B | 278 B | ✓ |
| Overall change | 7.88 kB | 7.88 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages Modern</strong></summary>

|  | zeit/next.js canary  | lfades/next.js examples/cms-contentful | Change |
| - | - | - | - |
| _app.module.js gzip | 598 B | 598 B | ✓ |
| _error.module.js gzip | 2.09 kB | 2.09 kB | ✓ |
| hooks.module.js gzip | 383 B | 383 B | ✓ |
| index.module.js gzip | 223 B | 223 B | ✓ |
| link.module.js gzip | 1.52 kB | 1.52 kB | ✓ |
| routerDirect..dule.js gzip | 279 B | 279 B | ✓ |
| withRouter.m..dule.js gzip | 278 B | 278 B | ✓ |
| Overall change | 5.37 kB | 5.37 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | zeit/next.js canary  | lfades/next.js examples/cms-contentful | Change |
| - | - | - | - |
| _buildManifest.js gzip | 61 B | 61 B | ✓ |
| _buildManife..dule.js gzip | 61 B | 61 B | ✓ |
| Overall change | 122 B | 122 B | ✓ |

</details>

<details>
<summary><strong>Serverless bundles</strong></summary>

|  | zeit/next.js canary  | lfades/next.js examples/cms-contentful | Change |
| - | - | - | - |
| _error.js | 557 kB | 557 kB | ✓ |
| 404.html | 4.18 kB | 4.18 kB | ✓ |
| hooks.html | 3.81 kB | 3.81 kB | ✓ |
| index.js | 557 kB | 557 kB | ✓ |
| link.js | 595 kB | 595 kB | ✓ |
| routerDirect.js | 587 kB | 587 kB | ✓ |
| withRouter.js | 588 kB | 588 kB | ✓ |
| Overall change | 2.89 MB | 2.89 MB | ✓ |

</details>

</details>

File: examples/cms-contentful/README.md [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+- Use dummy data for the text.
+- You can write markdown for the **content** field.
+- For images, you can download them from [Unsplash](https://unsplash.com/).
+- Pick the **author** you created earlier.
+
+**Important:** For each entry, you need to click on **Publish**. If not, the entry will be in draft state.
+
+### Step 5. Set up environment variables
+
+From your contentful space, go to **Settings > API keys**, there should be an example API key, you can use that one or create a new one.
```
Comment (chibicode) [inline examples/cms-contentful/README.md:85]:
```suggestion
From your contentful space, go to **Settings > API keys**. There should be an example Content delivery / preview token - you can use this API key.
```

File: examples/cms-contentful/README.md [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+```bash
+cp .env.example .env
+```
+
+Then set each variable on `.env`:
+
+- `NEXT_EXAMPLE_CMS_CONTENTFUL_SPACE_ID` should be the **Space ID** field of your API Key
+- `NEXT_EXAMPLE_CMS_CONTENTFUL_ACCESS_TOKEN` should be the **Content Delivery API - access token** field of your API key
+- `NEXT_EXAMPLE_CMS_CONTENTFUL_PREVIEW_ACCESS_TOKEN` should be the **Content Preview API - access token** field of your API key
+- `NEXT_EXAMPLE_CMS_CONTENTFUL_PREVIEW_SECRET` should be any value you want, it must be URL friendly as the dashboard will send it as a query parameter to enable preview mode
```
Comment (chibicode) [inline examples/cms-contentful/README.md:98]:
```suggestion
- `NEXT_EXAMPLE_CMS_CONTENTFUL_PREVIEW_SECRET` should be any value you want. It must be URL friendly as the dashboard will send it as a query parameter to enable preview mode
```

File: examples/cms-contentful/README.md [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+
+yarn install
+yarn dev
+```
+
+Your blog should be up and running on [http://localhost:3000](http://localhost:3000)! If it doesn't work, post on [GitHub discussions](https://github.com/zeit/next.js/discussions).
+
+### Step 7. Try preview mode
+
+On your contentful space, go to **Settings > Content preview**, then add a new content preview for development, fill the form like so:
```
Comment (chibicode) [inline examples/cms-contentful/README.md:125]: Will this be added later?

File: examples/cms-contentful/README.md [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+
+Now, let's add a new content preview for development, the **Name** field may be anything, like `Development`, then under **Content preview URLs**, check **post** and set its input to:
+
+```
+http://localhost:3000/api/preview?secret=<NEXT_EXAMPLE_CMS_CONTENTFUL_PREVIEW_SECRET>&slug={entry.fields.slug}
+```
+
+Replace `<NEXT_EXAMPLE_CMS_CONTENTFUL_PREVIEW_SECRET>` with its respective value in `.env`
```
Comment (chibicode) [inline examples/cms-contentful/README.md:133]:
```suggestion
Replace `<NEXT_EXAMPLE_CMS_CONTENTFUL_PREVIEW_SECRET>` with its respective value in `.env`.
```

File: examples/cms-contentful/README.md [hunk 5]
```diff
@@ -0,0 +1,158 @@
...
+
+yarn install
+yarn dev
+```
+
+Your blog should be up and running on [http://localhost:3000](http://localhost:3000)! If it doesn't work, post on [GitHub discussions](https://github.com/zeit/next.js/discussions).
+
+### Step 7. Try preview mode
+
+On your contentful space, go to **Settings > Content preview**, then add a new content preview for development, fill the form like so:
```
Comment (chibicode) [inline examples/cms-contentful/README.md:125]:
```suggestion
On your contentful space, go to **Settings > Content preview**, then add a new content preview for development.
```

File: examples/cms-contentful/README.md [hunk 6]
```diff
@@ -0,0 +1,158 @@
...
+
+Now, let's add a new content preview for development, the **Name** field may be anything, like `Development`, then under **Content preview URLs**, check **post** and set its input to:
```
Comment (chibicode) [inline examples/cms-contentful/README.md:127]:
```suggestion
The **Name** field may be anything, like `Development`. Then, under **Content preview URLs**, check **post** and set its value to:
```

File: examples/cms-contentful/README.md [hunk 7]
```diff
@@ -0,0 +1,158 @@
...
+```
+http://localhost:3000/api/preview?secret=<NEXT_EXAMPLE_CMS_CONTENTFUL_PREVIEW_SECRET>&slug={entry.fields.slug}
+```
+
+Replace `<NEXT_EXAMPLE_CMS_CONTENTFUL_PREVIEW_SECRET>` with its respective value in `.env`.
+
+Once saved, go to one of the posts you've created and:
+
+- **Update the title**. For example, you can add `[Draft]` in front of the title.
+- The state of the post will switch to **CHANGED**, **do not** publish it. By doing this, the post will be in draft state.
```
Comment (chibicode) [inline examples/cms-contentful/README.md:138]:
```suggestion
- The state of the post will switch to **CHANGED** automatically. **Do not** publish it. By doing this, the post will be in draft state.
```

**Review by chibicode — APPROVED** [review]

**Pull request merged**